src/utils/map-controls.js

```diff
@@ -108,10 +108,10 @@ export default class MapControls {
     const oldViewport = this.mapState.getViewportProps();
     const newViewport = newMapState.getViewportProps();
 
-    if (this.onChangeViewport &&
+    if (this.onViewportChange &&
```

Comment:
Q: Should we even try to keep the names identical here? This seems like a rather different class...

This code has changed a lot so I might not have the full picture. If my comment does not make sense, just ignore it.